src/lib.rs

```diff
@@ -383,57 +478,104 @@ impl<T> Receiver<T> {
                 std::thread::sleep(std::time::Duration::from_millis(10));
 
                 // Write our waker instance to the channel.
+                // SAFETY: we are still in the empty state, meaning that the sender will not try
+                // to access the waker until it sees the state set to RECEIVING
```

Comment:
We can't guarantee that the channel is in the `EMPTY` state when we write the waker here. It could also be `MESSAGE` or `DISCONNECTED` if the `Sender` sends or is dropped in between here. But I don't think you need to explain all of that. It's enough to explain that the `Sender` will not access the waker until it sees the `RECEIVING` state, which we are responsible for setting.